traffic-manager-application-gateway-demo-setup/azuredeployappgw.json

```diff
@@ -0,0 +1,342 @@
+      "metadata": {
+        "description": "Location of resources"
+      }
+    },
+    "appGwName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name for Application Gateway"
+      }
+    }
+  },
+  "variables": {
+    "applicationGatewayName": "[parameters('appGwName')]",
+    "publicIPAddressName": "[concat(parameters('appGwName'),'IP')]",
+    "virtualNetworkName": "[concat(parameters('appGwName'),'VNet')]",
+    "subnetName": "[concat(parameters('appGwName'),'Subnet')]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+    "applicationGatewayID": "[resourceId('Microsoft.Network/applicationGateways',variables('applicationGatewayName'))]",
+    "backendDnsPrefix": "[concat(parameters('appGwName'),'-backend-')]",
+    "wsDeploymentName1": "[concat(parameters('appGwName'),'ws1')]",
+    "wsDeploymentName2": "[concat(parameters('appGwName'),'ws2')]",
+    "appGatewayIPConfigName": "[concat(parameters('appGwName'),'-ipconfig')]",
+    "appGatewayrequestRoutingRuleName": "[concat(parameters('appGwName'),'-rule1')]",
+    "appGatewayHttpListenerName": "[concat(parameters('appGwName'),'-listener')]",
+    "appGatewayBackendHttpSettingsName": "[concat(parameters('appGwName'),'-httpsettings')]",
+    "appGatewayBackendPoolName": "[concat(parameters('appGwName'),'-backendpool')]",
+    "appGatewayFrontendPortName": "[concat(parameters('appGwName'),'-port')]",
+    "appGatewayFrontendIPConfigName": "[concat(parameters('appGwName'),'-ipconfig')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetPrefix": "10.0.0.0/28",
+    "webServerTemplateLocation": "https://raw.githubusercontent.com/AlekseiPolkovnikov/azure-quickstart-templates/tmagw/traffic-manager-application-gateway-demo-setup/",
+    "webServerTemplateName": "azuredeploywebserver.json",
+    "server1TestPage": "[concat('<h1 style=\"color:red;font-size:300%;\">This is Server 1, Location: ', parameters('location'), '</h1>')]",
+    "server2TestPage": "[concat('<h1 style=\"color:blue;font-size:300%;\">This is Server 2, Location: ', parameters('location'), '</h1>')]",
+    "serverTestPageInfo": "<p>Send next request. If Cookie-based affinity is enabled, clear the cookies to change the backend server.</p><p><strong>Request headers:</strong> <br /><?php $hs = apache_request_headers();foreach($hs as $h => $value){echo \"$h: $value <br />\n\";}?></p>"
+  },
+  "outputs": {
+    "ipId": {
+      "type": "string",
+      "value": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddressName'))]"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2015-01-01",
+      "name": "[variables('wsDeploymentName1')]",
+      "type": "Microsoft.Resources/deployments",
```

Comment:
Same as above. Do we consider the overhead worthwhile for 2 instances? How about return params?